### PR TITLE
CP-12967: Change xcp-rrdd-iostat to produce a new RRD for number of tapdisks in low memory mode

### PR DESCRIPTION
**Pull request by koushikcgit**
Added new RRD for number of tapdisks in low memory mode

Signed-off-by: Koushik Chakravarty <koushik.chakravarty@citrix.com>

### CODE REVIEW

**Comment by xen-git** [comment]
Can one of the admins verify this patch for testing?

File: rrdp_iostat.ml [hunk 1]
```diff
@@ -614,6 +615,23 @@ let gen_metrics () =
...
 		) sr_to_stats_values in
 
+	(* Get the blktap3 stats and iterate the stats list to count the
+		number of tapdisks in low memory mode *)
+	let data_sources_low_mem_mode =
+		let count = ref 0L in
+		let (++) = Int64.add in 
+		List.iter (fun ((domid, devid), stats) ->
+			if stats.st_mem_mode then count := !count ++ 1L
+		)domid_devid_to_stats_blktap3;
```
Comment (johnelse) [inline rrdp_iostat.ml]: With a fold you could dispense with the ref, e.g.:

```
let count = List.fold
	(fun acc ((_, _), stats) ->	if stats.st_mem_mode then acc + 1 else acc)
	domid_devid_to_stats_blktap3
in
...
```

**Comment by koushikcgit** [comment]
@johnelse Updated based on your suggestion to use List.fold

**Comment by johnelse** [comment]
@xen-git test this please

**Comment by johnelse** [comment]
Patch looks good. Let's see what @xen-git thinks.

**Pull request merged**